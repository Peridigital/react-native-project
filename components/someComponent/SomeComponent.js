import React from 'react'
import { noop } from 'lodash'
import {
  Pressable,
  Text,
  View
} from 'react-native'
import PropTypes from 'prop-types'
import style from './SomeComponent.style'

export function SomeComponent ({ isActive, label, toggleIsActive }) {
  return (
    <View style={style.container}>
      <Pressable
        onPress={toggleIsActive}
        style={style.button}
      >
        <Text>Toggle isActive</Text>
      </Pressable>
      <Text>
        {label}
      </Text>
      <Text>
        Current state is: {isActive ? 'active' : 'inactive'}
      </Text>
    </View>
  )
}

SomeComponent.propTypes = {
  isActive: PropTypes.bool,
  label: PropTypes.string,
  toggleIsActive: PropTypes.func
}

SomeComponent.defaultProps = {
  isActive: false,
  label: 'hello world',
  toggleIsActive: noop
}

export default SomeComponent
