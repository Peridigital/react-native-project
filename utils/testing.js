import ShallowRenderer from 'react-test-renderer/shallow'

/**
 * Creates a shallow rendered string representation of a given react component
 *
 * @param {any} component - A React component to be snapshotted ex: <Text>Hello World<Text/>
 *
 * @returns {string} A shallow rendered string representation of the React component
 */
export function shallow (component) {
  const renderer = new ShallowRenderer()
  renderer.render(component)
  return renderer.getRenderOutput()
}
