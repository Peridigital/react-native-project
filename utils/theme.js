const colors = {
  primaryBackground: '#0C0D12',
  secondaryBackground: '#151821',
  primaryText: '#EAEAEA',
  border: '#7E4033',
  borderHighlight: '#D0A499'
}

export default colors
