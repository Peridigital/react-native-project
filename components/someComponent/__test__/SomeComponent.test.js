import React from 'react'
import { describe, it, expect } from 'jest'
import { shallow } from '../../../utils/testing'
import SomeComponent from '../SomeComponent'

describe('SomeComponent', () => {
  it('should render', () => {
    const wrapper = shallow(<SomeComponent />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render when active', () => {
    const wrapper = shallow(<SomeComponent isActive={true} />)

    expect(wrapper).toMatchSnapshot()
  })
})
