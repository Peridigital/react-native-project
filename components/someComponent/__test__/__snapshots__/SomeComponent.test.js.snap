// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomeComponent should render 1`] = `
<View
  style={
    {
      "gap": 15,
    }
  }
>
  <Memo(Pressable)
    onPress={[Function]}
    style={
      {
        "backgroundColor": "orange",
        "padding": 10,
      }
    }
  >
    <Text>
      Toggle isActive
    </Text>
  </Memo(Pressable)>
  <Text>
    hello world
  </Text>
  <Text>
    Current state is: 
    inactive
  </Text>
</View>
`;

exports[`SomeComponent should render when active 1`] = `
<View
  style={
    {
      "gap": 15,
    }
  }
>
  <Memo(Pressable)
    onPress={[Function]}
    style={
      {
        "backgroundColor": "orange",
        "padding": 10,
      }
    }
  >
    <Text>
      Toggle isActive
    </Text>
  </Memo(Pressable)>
  <Text>
    hello world
  </Text>
  <Text>
    Current state is: 
    active
  </Text>
</View>
`;
